### PR TITLE
Implement python wrappers for predictions and tune speed

### PR DESCRIPTION
**Pull request by Bowid**
1. Predict now use std::set instead priority_queue. From my point of
view it is better to maintain set of fixed size and drop predictions
with too low score immideately instead storing them in ordered manner.
2. Implemented extension class of StarSpace to make things a bit more
pythonic. It is definitely better to have return value instead of
passing parameters to fill to functions. From my point of view, speed is
not changed in case modern C++ compilers handle return of complex values
properly without copying them.

### CODE REVIEW

**Comment by pmirla** [comment]
Hello Bowid. I am also working on python wrapper. Can you please write back with your contacts to pavanmirla @gmail. Thanks

**Comment by Bowid** [comment]
> Hello Bowid. I am also working on python wrapper. Can you please write back with your contacts to pavanmirla @gmail. Thanks

Replied. Also here is the sample, let it be here

```python
import starwrap

model_path = '/path/to/trained/model'

args = starwrap.args()
model = starwrap.starSpace(args)
print(model_path)
model.initFromSavedModel(model_path)

line = input("Enter text for classification: ")
parsed = model.parseDoc(line, ' ')

# Output is list of tuples in format (probability, token_id)
# Predict takes two arguments: parsed doc and maximum predictions count
tokens = model.predict(parsed, 10)

probabilities = [item[0] for item in tokens]
# Token_id is integer, thus it should be rendered to feature (string)
features = model.renderTokens(tokens)

for feature, probability in zip(features, probabilities):
    print('Feature {f} probability {p}'.format(f=feature, p=probability))
```

**Comment by Bowid** [comment]
I have some more thoughts to make things more pythonic, but it is nice to have feedback from FB guys and probably merge this PR to master.

File: src/starspace.cpp [hunk 1]
```diff
@@ -299,19 +300,44 @@ void StarSpace::loadBaseDocs() {
 void StarSpace::predictOne(
     const vector<Base>& input,
     vector<Predictions>& pred) {
+      predict(input, pred, args_->K);
+}
+
+void StarSpace::predict(
```
Comment (ledw) [inline src/starspace.cpp]: Could you add some test around this function?

**Review by ledw — COMMENTED** [review]
Looks good over all. Thanks for contributing! 
Please resolve the conflict with current version. Please run example tests to make sure this does not break current system, and add some test around the new predict function.

File: src/starspace.cpp (hunk 1 above)
Comment (Bowid) [reply src/starspace.cpp]: Whoops, unfortunately things goes wrong and i closed this PR after merging changes. So, the new one is here https://github.com/facebookresearch/StarSpace/pull/279

Tests are coming soon

**Pull request closed without merging**